test/common/index.js

```diff
@@ -709,6 +709,11 @@ exports.expectsError = function expectsError(fn, settings, exact) {
       assert(error instanceof type,
              `${error.name} is not instance of ${type.name}`);
     }
+    if ('code' in settings) {
```

Comment:
This is a very good idea! I think L703 agrees with your train of thought 😉  
https://github.com/nodejs/node/blob/592766be901381a12870fa780ddee98e97b8a84a/test/common/index.js#L703